lib/connextclient/ConnextClient.ts

```diff
@@ -721,18 +757,22 @@ class ConnextClient extends SwapClient {
         mergeMap(() => {
           // use defer to only create the inner observable when the outer one subscribes
           return defer(() => {
-            return from(
-              this.sendRequest('/request-collateral', 'POST', {
-                assetId,
-                amount: (minCollateralRequestUnits.toLocaleString('fullwide', { useGrouping: false })),
-              }),
-            );
+            const requestCollateralPromise = this.sendRequest('/request-collateral', 'POST', {
+              assetId,
+              amount: (minCollateralRequestUnits.toLocaleString('fullwide', { useGrouping: false })),
+            }).finally(() => {
+              this.requestCollateralPromises.delete(currency);
+            });
+
+            this.requestCollateralPromises.set(currency, requestCollateralPromise);
+            return from(requestCollateralPromise);
```

Comment:
Maybe we can also remove the above lines and replace it with `return from(this.requestCollateralInBackground(currency, minCollateralRequestUnits))` as it seems to be doing the same thing (with extra logging)?

Reply:
`return from(this.requestCollateralInBackground(currency, minCollateralRequestUnits))` doesn't work exactly because requestCollateralInBackground doesn't return a promise, by design the promise is resolved/handled in the background. I could just call the method and return nothing? that should be fine right?